### ISSUE TITLE
Support Themes - via Class using `setvar` Template Tag

### ISSUE THREAD

**Issue by tacc-wbomar** [TACC/Core-CMS#57]
In GitLab by @tacc-wbomar on Dec 3, 2020, 17:14

# Alternative

- #191

# Goal

1. Support dynamic addition of a `t-____` class to the `<body>`.
2. Support generic API for CMS projects to add this class.

# Requirements

1. Each project __must__ be able to add one or more theme classes.
1. There __must__ be a way to use logic to determine theme class.
1. The logic __should__ be defined per CMS.

# Direction

## Goal

1. To add class:
    - Use template logic to add class.
2. API ideas:
    - Use template logic in `cms-site-resources` per project.
    - Add content to secrets/settings, Python logic to get value, pass data to template.

## Code

Example template logic is:

### `taccsite_cms/settings.py`
```
- {% load cms_tags staticfiles sekizai_tags cache i18n %}
+ {% load cms_tags staticfiles sekizai_tags cache i18n custom_utility_tags %}
  <!doctype html>
  <html>

    {% endblock custom_css %}
  </head>

- <body class="o-site  {% block page_type_class %}{% endblock page_type_class %}">
+ {# Get theme for current page #}
+ {# NOTE: This logic should be in Python #}
+ {# TODO: Remove (or migrate to theme) upon 2020 launch #}
+ {% if '2020' in request.path %}{% setvar "2020" as themeName %}{% endif %}
+ {% if '2019' in request.path %}{% setvar "2019" as themeName %}{% endif %}
+ {% if '2018' in request.path %}{% setvar "2018" as themeName %}{% endif %}
+ 
+ <body class="t-{{themeName}}  o-site  {% block page_type_class %}{% endblock page_type_class %}">
    {% cms_toolbar %}

    <!-- Custom Navigation Block per Site to Customize the Site Header Branding. -->
```

A `setvar` tag can be created via:

### `taccsite_cms/settings.py`
```
              ],
              'libraries': {
                  'custom_portal_settings': 'taccsite_cms.templatetags.custom_portal_settings',
+                 'custom_utility_tags': 'taccsite_cms.templatetags.custom_utility_tags',
              },
              'loaders': [
                  'django.template.loaders.filesystem.Loader',
```

### `taccsite_cms/templatetags/custom_utility_tags.py`
```
from django import template
register = template.Library()

#
# WARNING: Not recommended, but a common request
# SEE: https://stackoverflow.com/a/1070575
@register.simple_tag
def setvar(val=None):
    """
    Custom Template Tag `setvar`

    Use: Set Variables at Will in Templates.

    Load custom tag into template:
        {% load custom_generic_tags %}

    Template inline usage:
        {{ setvar "Foo" as foo }}

    Example:
        {% load custom_template_tags %}

        {% if is_login %}
            {% setvar "Save" as action %}
        {% else %}
            {% setvar "Compile" as action %}
        {% endif %}

        Would you like to {{action}} this code?
    """
    return val

```

**Comment by tacc-wbomar** [comment]
Closed in favor of alternative, #191.

**Comment by tacc-wbomar** [comment]
Closed in favor of alternative, #191.
